### PR TITLE
Create docker-compose.yml

### PR DESCRIPTION
**Pull request by kmehant**
Fixes #249

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: "3"
+services:
+  web:
+    build:
+      context: ui/
+      dockerfile: Dockerfile
```
Comment (agentmilindu) [inline docker-compose.yml]: This line is not needed

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      ports:
+        - "3000:3000"
+      labels:
+        com.example.description: "Docker container for Bassa UI"
+  api:
+    build:
+      context: components/core
+      dockerfile: Dockerfile
```
Comment (agentmilindu) [inline docker-compose.yml]: This line is not needed.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        - "8000:8000"
+      labels:
+        com.example.description: "Docker container for Bassa backend"
+  db:
+    image: mysql
+    restart: always
+    ports:
+      - 6603:3306
+    environment:
+      MYSQL_ROOT_PASSWORD: ${BASSA_DB_PASSWORD}
```
Comment (agentmilindu) [inline docker-compose.yml]: Let's add `BASSA_DB_USERNAME` elso.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+version: "3"
+services:
+  web:
+    build:
+      context: ui/
```
Comment (agentmilindu) [inline docker-compose.yml]: Let's remove both `context` and `dockerfile` and keep the `build`
```
web:
  build: .
```

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      dockerfile: Dockerfile
+      ports:
+        - "3000:3000"
+      labels:
+        com.example.description: "Docker container for Bassa UI"
+  api:
+    build:
+      context: components/core
```
Comment (agentmilindu) [inline docker-compose.yml]: Let's remove both `context` and `dockerfile` and keep the `build`
```
web:
  build: .
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      dockerfile: Dockerfile
+      ports:
+        - "8000:8000"
+      labels:
+        com.example.description: "Docker container for Bassa backend"
+  db:
+    image: mysql
+    restart: always
+    ports:
+      - 6603:3306
```
Comment (agentmilindu) [inline docker-compose.yml]: Any specific reason to change the port?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        - "3000:3000"
+      labels:
+        com.example.description: "Docker container for Bassa UI"
+  api:
+    build:
+      context: components/core
+      dockerfile: Dockerfile
+      ports:
+        - "8000:8000"
+      labels:
```
Comment (agentmilindu) [inline docker-compose.yml]: Let's add the following part also,
```
depends_on:
      - "db"
```

File: docker-compose.yml (hunk 3 above)
Comment (agentmilindu) [reply docker-compose.yml]: Let's do as following, 

```
MYSQL_DATABASE: ${BASSA_DB_NAME}
MYSQL_USER: ${BASSA_DB_USERNAME}
MYSQL_PASSWORD: ${BASSA_DB_PASSWORD} 
```

and remove `MYSQL_ROOT_PASSWORD: ${BASSA_DB_PASSWORD}`

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        com.example.description: "Docker container for Bassa backend"
+  db:
+    image: mysql
+    restart: always
```
Comment (agentmilindu) [inline docker-compose.yml]: We need to set volumes also, 
```
volumes:
      - data-volume:/var/lib/mysql
```

**Comment by agentmilindu** [comment]
At the end of the file, add the following, with the service's level,

```
volumes:
  data-volume:
```

Refer: https://docs.docker.com/compose/compose-file/#volume-configuration-reference

**Comment by kmehant** [comment]
@agentmilindu addressed the changes :) kindly review

File: components/core/setup.sh [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+echo "Installing mysql-server"
+apt-get install -y mysql-server
+echo "mysql-server installed"
+
+echo "Installing libmysqlclient"
+apt-get install libmysqlclient-dev
+echo "libmysqlclient installed"
+
+echo "Installing node"
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
```
Comment (agentmilindu) [inline components/core/setup.sh]: Let's remove the `sudo` here also. 

eg: `curl -sL https://deb.nodesource.com/setup_6.x | bash -`

**Pull request closed without merging**